### PR TITLE
qa: Fix traceback during fs cleanup between tests

### PR DESCRIPTION
**Pull request by kotreshhr**
The 'mon_allow_pool_delete' option is set to 'True'
in 'setUp' of 'TestVolumes' and is cleared of in
corresponding 'tearDown' function. Hence, any pool
deletion in parent classes such as 'CephFSTestCase'
would fail. This patch fixes the same by setting
the config 'mon_allow_pool_delete' option in the
'CephFSTestCase'.

Fixes: https://tracker.ceph.com/issues/46597
Signed-off-by: Kotresh HR <khiremat@redhat.com>

## Checklist
- [ ] References tracker ticket
- [ ] Updates documentation if necessary
- [ ] Includes tests for new functionality or reproducer for bug

---

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard backend`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`

</details>

### CODE REVIEW

File: qa/tasks/cephfs/cephfs_test_case.py [hunk 1]
```diff
@@ -175,6 +181,10 @@ def tearDown(self):
         for i, m in enumerate(self.mounts):
             m.client_id = self._original_client_ids[i]
 
+        if self.vol_created:
+            self.mgr_cluster.mon_manager.raw_cluster_cmd("fs", "volume", "rm",
```
Comment (batrick) [inline qa/tasks/cephfs/cephfs_test_case.py:185]: I don't really want to mix the raw fs commands with the volumes plugin. Let's keep using the raw `ceph fs` commands outside of the `test_volumes.py` code.

File: qa/tasks/cephfs/cephfs_test_case.py [hunk 2]
```diff
@@ -63,6 +63,8 @@ class CephFSTestCase(CephTestCase):
     def setUp(self):
         super(CephFSTestCase, self).setUp()
 
+        self.config_set('mon', 'mon_allow_pool_delete', True)
```
Comment (batrick) [inline qa/tasks/cephfs/cephfs_test_case.py:66]: This looks like the real fix. Why do we need the other changes? (a separate commit + explanation would be helpful)

File: qa/tasks/cephfs/cephfs_test_case.py (hunk 1 above)
Comment (kotreshhr) [reply qa/tasks/cephfs/cephfs_test_case.py]: Done.

File: qa/tasks/cephfs/cephfs_test_case.py (hunk 2 above)
Comment (kotreshhr) [reply qa/tasks/cephfs/cephfs_test_case.py:66]: Yes, that's correct. This is sufficient to fix the traceback.
The other changes doesn't have functional impact. I have split the commit into two and tried explaining there. Hope that helps.

File: qa/tasks/cephfs/test_volumes.py [hunk 3]
```diff
@@ -141,15 +141,6 @@ def _generate_random_clone_name(self, count=1):
     def _enable_multi_fs(self):
         self._fs_cmd("flag", "set", "enable_multiple", "true", "--yes-i-really-mean-it")
 
-    def _create_or_reuse_test_volume(self):
```
Comment (kotreshhr) [inline qa/tasks/cephfs/test_volumes.py:144]: Reviewers Note: This is a dead code that's removed as REQUIRE_FILESYSTEM flag is True by default and hence 'CephFSTestCase' creates the fs. Also correspond fs removal in tearDown is dead code.

File: qa/tasks/cephfs/cephfs_test_case.py [hunk 4]
```diff
@@ -127,8 +129,12 @@ def setUp(self):
             if ent_type == "client" and ent_id not in client_mount_ids and ent_id != "admin":
                 self.mds_cluster.mon_manager.raw_cluster_cmd("auth", "del", entry['entity'])
 
+        self.volname = None
+        self.vol_created = False
```
Comment (batrick) [inline qa/tasks/cephfs/cephfs_test_case.py:133]: I think it's sufficient to just test if `self.fs is None` later so I think we can remove the above two variables.

File: qa/tasks/cephfs/cephfs_test_case.py [hunk 5]
```diff
@@ -175,6 +181,9 @@ def tearDown(self):
         for i, m in enumerate(self.mounts):
             m.client_id = self._original_client_ids[i]
 
+        if self.vol_created:
+            self.fs.rm(self.volname)
```
Comment (batrick) [inline qa/tasks/cephfs/cephfs_test_case.py:185]:
```suggestion
        if self.fs is not None:
            self.fs.rm()
            self.fs = None
```

File: qa/tasks/cephfs/filesystem.py [hunk 6]
```diff
@@ -611,6 +613,15 @@ def create(self):
 
         self.getinfo(refresh = True)
 
+    def rm(self, name='cephfs'):
```
Comment (batrick) [inline qa/tasks/cephfs/filesystem.py:616]: @rishabh-d-dave is working on a similar method here:

https://github.com/ceph/ceph/pull/32581/files#diff-b3a56ab86d2e46ef827df2b8b21fc142R585

Suggest you pull this refactor commit out into a separate PR that blocks on the merge of #32581.

Comment (kotreshhr) [reply qa/tasks/cephfs/filesystem.py]: Yep, makes sense. Done.

File: qa/tasks/cephfs/cephfs_test_case.py (hunk 4 above)
Comment (kotreshhr) [reply qa/tasks/cephfs/cephfs_test_case.py]: self.volname is heavily used in 'TestVolumes' (test_volumes.py).

File: qa/tasks/cephfs/cephfs_test_case.py (hunk 5 above)
Comment (kotreshhr) [reply qa/tasks/cephfs/cephfs_test_case.py]: Addressed in https://github.com/ceph/ceph/pull/36368

File: qa/tasks/cephfs/cephfs_test_case.py (hunk 4 above)
Comment (kotreshhr) [reply qa/tasks/cephfs/cephfs_test_case.py]: Addressed in https://github.com/ceph/ceph/pull/36368

**Comment by batrick** [comment]
jenkins test dashboard backend

**Comment by batrick** [comment]
https://pulpito.ceph.com/?branch=wip-pdonnell-testing-20200730.193838

Failures known/unrelated.
